com.unity.sg2/Editor/GraphUI/DataModel/Blackboard/VariableSettings.cs

```diff
@@ -0,0 +1,144 @@
+
+        public static readonly VariableSetting<ColorMode> colorMode = new(
+            label: "Mode",
+            getter: model =>
+            {
+                var isHdr = model.ContextEntry.GetField<bool>(kIsHdr)?.GetData() ?? false;
+                return isHdr ? ColorMode.HDR : ColorMode.Default;
+            },
+            setter: (model, value) =>
+            {
+                var field = model.ContextEntry.GetField<bool>(kIsHdr) ??
+                    model.ContextEntry.AddField(kIsHdr, false);
+
+                field.SetData(value is ColorMode.HDR);
+            });
+
+        #endregion
+
+        #region Sampler State
+
+        public static readonly VariableSetting<SamplerStateType.Filter> samplerStateFilter =
+            CreateFromSubField<SamplerStateType.Filter>(model => model.ContextEntry.GetTypeField(), SamplerStateType.kFilter, "Filter");
+
+        public static readonly VariableSetting<SamplerStateType.Wrap> samplerStateWrap =
+            CreateFromSubField<SamplerStateType.Wrap>(model => model.ContextEntry.GetTypeField(), SamplerStateType.kWrap, "Wrap");
+
+        #endregion
+
+        #region Shared
+
+        public static readonly VariableSetting<DataSource> shaderDeclaration =
+            CreateFromContextEntryTag<DataSource>(kDataSource, "Shader Declaration");
+
+        #endregion
+
+        static VariableSetting<T> CreateFromContextEntryTag<T>(string key, string label, T defaultValue = default) =>
+            new(
+                label: label,
+                getter: model =>
+                {
+                    var field = model.ContextEntry.GetField<T>(key);
+                    return field == null ? defaultValue : field.GetData();
+                },
+                setter: (model, value) =>
+                {
+                    (model.ContextEntry.GetField<T>(key) ?? model.ContextEntry.AddField<T>(key)).SetData(value);
+                });
+
+        static VariableSetting<T> CreateFromSubField<T>(
+            Func<GraphDataVariableDeclarationModel, FieldHandler> fieldProvider,
+            string key,
+            string label,
+            T defaultValue = default) =>
+            new(
+                label: label,
+                getter: model =>
+                {
+                    var field = fieldProvider.Invoke(model);
+                    var subField = field.GetSubField<T>(key);
+                    return subField != null ? subField.GetData() : defaultValue;
+                },
+                setter: (model, value) =>
+                {
+                    var field = fieldProvider.Invoke(model);
+                    var subField = field.GetSubField<T>(key);
+                    if (subField != null)
+                    {
+                        subField.SetData(value);
+                    }
+                    else
+                    {
+                        field.AddSubField(key, value);
+                    }
+                });
+    }
```

Comment:
Same here-- a way to consolidate this so that the type describes its own settings along with its behavior-- so there is one point of entry/source of truth for how we can expect a type to be interacted with.

No change requested, just an observation that we should make the type system more robust in creating C# representations that more transparently work with the rest of Unity, the UI, and can be trivially translatable to CLDS.